l10n_it_account_stamp/views/product_view.xml

```diff
@@ -11,11 +11,12 @@
                 <group name='group_standard_price' position="after">
                     <group string="Tax stamp">
                         <field name="is_stamp"/>
-                        <p attrs="{'invisible': [('is_stamp', '=', False)]}" colspan="2">
+                        <field name="auto_compute"  attrs="{'invisible': [('is_stamp', '=', False)]}"/>
+                        <p attrs="{'invisible': [('auto_compute', '=', False)]}" colspan="2">
                             For each invoice, the base amount for each selected tax will be added up and used to determine the application of the account stamp.
                         </p>
-                        <field name="stamp_apply_min_total_base" attrs="{'invisible': [('is_stamp', '=', False)], 'required': [('is_stamp', '=', True)]}"/>
-                        <field name="stamp_apply_tax_ids" attrs="{'invisible': [('is_stamp', '=', False)]}" widget="many2many_tags">
+                        <field name="stamp_apply_min_total_base" attrs="{'invisible': [('auto_compute', '=', False)], 'required': [('auto_compute', '=', True)]}"/>
+                        <field name="stamp_apply_tax_ids" attrs="{'invisible': [('auto_compute', '=', False)]}" widget="many2many_tags">
```

Comment:
Deselecting 'is stamp' does not hide stamp data:

![peek 2019-01-23 18-06](https://user-images.githubusercontent.com/1033131/51623795-c6c66b00-1f39-11e9-804a-a8ea9b4d9a41.gif)

Reply:
Fixed.
Just wondering...
Do you think we could add ```'required': [('auto_compute', '=', True)]```
to ```stamp_apply_tax_ids```?
(e.g. maybe trouble with default product)

Reply:
@primes2h scusami se arrivo solo ora, quindi il bollo viene aggiunto in automatico sulle fatture se imposto l'auto_compute, mentre in caso contrario me lo devo ricordare.

La logica del bottone sarebbe stata di far vedere all'utente che ci sarebbe da applicare il bollo, e quindi decidere se addebitarlo o no. Inoltre decidere se pagarlo o proprio no (mettendolo nell'xml) togliendolo con il flag.

Così non si dà più alcuna scelta all'utente per fattura, ma solo una scelta fissata a priori.
Per me non è corretto, magari non ho chiaro quello che volevi fare.

Reply:
> @primes2h scusami se arrivo solo ora, quindi il bollo viene aggiunto in automatico sulle fatture se imposto l'auto_compute, mentre in caso contrario me lo devo ricordare.

Ho aggiunto apposta l'immagine del bollo, se non lo vedo significa che non è stato applicato (sia nella configurazione automatica che quella manuale).


> La logica del bottone sarebbe stata di far vedere all'utente che ci sarebbe da applicare il bollo, e quindi decidere se addebitarlo o no.

L'inghippo dell'implementazione attuale IMHO è che quando io utente vedo che c'è un flag "imposta di bollo" mi aspetto che selezionandolo me lo attivi a prescindere.
Non dovrei preoccuparmi del fatto che sotto c'è un calcolo automatico, e che se non sto attento me lo disabilita quando non me ne accorgo.

Ho pensato alla separazione tra automatico e manuale per rendere più funzionale la cosa lato utente.
Se parto da un'installazione standard so che il modulo bollo funziona in automatico con soglia di 77 e rotti usando il prodotto predefinito (IMHO copre più del 90 % dei casi).

Se già in partenza so che voglio tenere sotto controllo tutto e decidere io quando applicare il bollo, imposto il prodotto bollo come manuale (disattivando il flag auto-compute) oppure mi creo un nuovo prodotto bollo di tipo manuale.

In questa configurazione potresti definire più prodotti bollo con soglie diverse automatiche e un prodotto bollo di tipo manuale e decidere quale impostare.


Ci ho ragionato tanto su questa cosa.
Penso che il problema di fondo sia il fatto che attualmente il flag dà la falsa impressione di rendere manuale l'applicazione (e quindi avere il controllo) di una cosa che non lo è di base.
Facendo clic sul flag infatti non viene disabilitato il calcolo automatico.

Comprendo bene che l'intento attuale sia quello di fornire flessibilità e sono totalmente d'accordo con te, ma questo imho non dovrebbe andare a scapito della semplicità d'uso.

Forse si potrebbero mettere d'accordo le due cose eliminando la scelta automatico/manuale a livello di prodotto e portandola a livello fattura, trasformando l'attuale flag "bollo"  in una scelta tra calcolo automatico e manuale.

Reply:
> Così non si dà più alcuna scelta all'utente per fattura, ma solo una scelta fissata a priori

@sergiocorato mi pare che l'utente possa sempre fare tutte le scelte.
Puoi descrivere il problema per passi?

Reply:
@primes2h @eLBati ho verificato meglio sul runbot e mi pare funzionale, al momento purtroppo non posso approfondire di più
:+1: